### PR TITLE
instalation: updates pytest and pytest-reana

### LINKED ISSUE #186: tests: use mock api client to write ping command test
Use the [`mock_api_client`](https://github.com/reanahub/reana-pytest-commons/blob/master/reana_pytest_commons/test_utils.py#L15) to write the [ping command test](https://github.com/reanahub/reana-client/blob/master/tests/test_cli_ping.py#L16-L20).

### PR DESCRIPTION
**Pull request by roksys**
Closes #186.

Signed-off-by: Rokas Maciulaitis <rokas.maciulaitis@cern.ch>

### CODE REVIEW

File: tests/test_cli_files.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+# This file is part of REANA.
+# Copyright (C) 2017, 2018 CERN.
+#
+# REANA is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""REANA client files tests."""
+
+import hashlib
+import io
```
Comment (dinosk) [inline tests/test_cli_files.py]: Not sure if this is wip, commenting on a few lines that popped up on my editor:
io is unused

File: tests/test_cli_files.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        }
+    ]
+    reana_token = '000000'
+    env = {'REANA_SERVER_URL': 'localhost'}
+    mocked_api_client = mock_base_api_client(status_code,
+                                             response,
+                                             'reana-server')
+    config = Config(mocked_api_client)
+    runner = CliRunner(env=env)
+    environment = {'REANA_SERVER_URL': 'localhost'}
```
Comment (dinosk) [inline tests/test_cli_files.py]: `environment` is unused

File: tests/test_cli_files.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    response = 'Content of file to download'
+    response_md5 = hashlib.md5(response.encode('utf-8')).hexdigest()
+    file = 'dummy_file.txt'
+    message = 'File {0} downloaded to'.format(file)
+    mocked_api_client = mock_base_api_client(status_code,
+                                             response,
+                                             'reana-server')
+    config = Config(mocked_api_client)
+    runner = CliRunner(env=env)
+    environment = {'REANA_SERVER_URL': 'localhost'}
```
Comment (dinosk) [inline tests/test_cli_files.py]: ^

File: tests/test_cli_files.py [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    env = {'REANA_SERVER_URL': 'localhost'}
+    file = 'dummy_file.txt'
+    response = [file]
+    message = 'was successfully uploaded.'
+    mocked_api_client = mock_base_api_client(status_code,
+                                             response,
+                                             'reana-server')
+    config = Config(mocked_api_client)
+    runner = CliRunner(env=env)
+    environment = {'REANA_SERVER_URL': 'localhost'}
```
Comment (dinosk) [inline tests/test_cli_files.py]: ^

File: tests/test_cli_ping.py [hunk 5]
```diff
@@ -9,10 +9,30 @@
 """REANA client ping tests."""
 
 from click.testing import CliRunner
+from mock import Mock
```
Comment (dinosk) [inline tests/test_cli_ping.py]: `Mock` is unused

File: tests/test_cli_ping.py [hunk 6]
```diff
@@ -9,10 +9,30 @@
...
 
-from reana_client.cli import cli
+from reana_client.api.client import Client
```
Comment (dinosk) [inline tests/test_cli_ping.py]: and `Client`

File: tests/test_cli_workflows.py [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""REANA client workflow tests."""
+
+import json
+
+from click.testing import CliRunner
+from mock import Mock
+
+from reana_client.api.client import Client
```
Comment (dinosk) [inline tests/test_cli_workflows.py]: `Mock` and `Client` in this file too

**Comment by dinosk** [comment]
Nice tests 👍

File: tests/test_cli_files.py (hunk 1 above)
Comment (roksys) [reply tests/test_cli_files.py]: Plugin that tracks unused variables/imports stoped working. Thanks for notice.

**Pull request merged**